bot.user.js

```diff
@@ -379,34 +385,24 @@ window.checkCollision = function(x, y, r) {
     if (window.visualDebugging) {
         window.drawDot(circle1.x, circle1.y, circle1.radius, 'blue', false);
     }
-    var shortest_distance = 10000;
     var avoid = false;
-    var circle2;
-
-    for (var snake in window.snakes) {
-        if (window.snakes[snake].nk != window.snake.nk) {
-            for (var y = window.snakes[snake].pts.length - 1; 0 <= y; y--) {
-                if (!window.snakes[snake].pts[y].dying) {
-                    var xx = window.snakes[snake].pts[y].xx + window.snakes[snake].fx;
-                    var yy = window.snakes[snake].pts[y].yy + window.snakes[snake].fy;
-                    circle2 = {
-                        x: xx,
-                        y: yy,
-                        radius: 15 * window.snakes[snake].sc * window.getScale()
-                    };
-                    if (window.circleIntersect(circle1, collisionScreenToCanvas(circle2))) {
-                        var distance = window.getDistance(window.getX(), window.getY(), xx, yy);
-                        if (distance < shortest_distance){
-                            window.changeGoalCoords(circle2);
-                            avoid = true;
-                            shortest_distance = distance;
-                        }
-                    }
-                }
-            }
-        }
-    }
-    return avoid;
+	if (window.collisionPoints[0] != null){
+		var circle2;
+		var multiplier = 1;
+		
+		circle2 = {
+			x: window.collisionPoints[0].xx + window.collisionPoints[0].fx,
+			y: window.collisionPoints[0].yy + window.collisionPoints[0].fy,
+			radius: 22 * collisionPoints[0].sc * window.getScale()
+		};
+		
+		if (window.circleIntersect(circle1, collisionScreenToCanvas(circle2))) {
+				window.changeGoalCoords(circle2, multiplier, (window.collisionPoints[0].sp > 10));
```

Comment:
My experience, after experimenting a bit with speeding, is that speeding to avoid another snake doesn't really work. Firstly, you lose a lot mass, secondly it takes more time to turn around, and thirdly you increase the risk of bumping into somethin. (At least for this type of naive usage. I guess it might make sense to speed up if trying to get away from a snake circling you and other situations where you are about to get trapped.) Have you checked that you gain anything by doing this?